app/pages/run/[runId].vue

```diff
@@ -24,11 +24,7 @@ const route = useRoute()
 const { data: initialRuns, pending: _pending, error: preloadedRunsError } = await useAsyncData('preloadedRuns', async (): Promise<Run[]> => {
   const { runId } = route.params
   if (typeof runId === 'string') {
-    const response = await fetch(`/api/runs/${runId}`)
-    if (!response.ok) {
-      throw new Error(`HTTP ${response.status}`)
-    }
-    const responseBody: ApiRun = await response.json()
+    const responseBody: ApiRun = await $fetch<ApiRun>(`/api/runs/${runId}`)
```

Comment:
I could be wrong, but do you need this since you're passing the type param?
```suggestion
    const responseBody = await $fetch<ApiRun>(`/api/runs/${runId}`)
```